File: 1957_delete_characters_to_make_fancy_string.java
class Solution {
    public String makeFancyString(String s) {
        int j = 2, length = s.length();
        if (length < 3) return s;
        char[] str = s.toCharArray();
        for (int i = 2; i < length; i++)
            if (str[j - 1] != str[i] || str[j - 2] != str[i])
                str[j++] = str[i];
        return new String(str, 0, j);
    }
}

File: 1957_delete_characters_to_make_fancy_string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        int j = 2, n = s.size();
        if (n < 3) return s;
        for (int i = 2; i < n; ++i) 
            if (s[j - 1] != s[i] || s[j - 2] != s[i]) 
                s[j++] = s[i];
        return s.substr(0, j);
    }
};
